Lib/test/test_itertools.py

```diff
@@ -590,6 +590,26 @@ def test_count_with_stride(self):
         self.assertEqual(type(next(c)), int)
         self.assertEqual(type(next(c)), float)
 
+    def test_count_threading(self, step=1):
+        # this test verifies multithreading consistency, which is
+        # mostly for testing builds without GIL, but nice to test anyway
+        count_to = 10_000
+        num_threads = 10
+        c = count(step=step)
+        def counting_thread():
+            for i in range(count_to):
+                next(c)
+        threads = []
+        for i in range(num_threads):
+            thread = threading.Thread(target=counting_thread)
+            thread.start()
+            threads.append(thread)
+        [thread.join() for thread in threads]
```

Comment:
Use something like:

```python
for thread in threads:
    thread.join()
```

It's an extra line, but I think it's the more common style and people tend to avoid using list comprehensions for side effects.